community/kernel/src/main/java/org/neo4j/kernel/impl/logging/StoreLogService.java

```diff
@@ -42,51 +41,96 @@
+            this.internalLogRotationDelay = internalLogRotationDelay;
+            this.maxInternalLogArchives = maxInternalLogArchives;
+            this.rotationExecutor = rotationExecutor;
+            return this;
+        }
+
+        public Builder withRotationListener( Consumer<LogProvider> rotationListener )
+        {
+            this.rotationListener = rotationListener;
+            return this;
+        }
+
+        public StoreLogService inStoreDirectory( FileSystemAbstraction fileSystem, File storeDir ) throws IOException
+        {
+            return toFile( fileSystem, new File( storeDir, INTERNAL_LOG_NAME ) );
+        }
+
+        public StoreLogService toFile( FileSystemAbstraction fileSystem, File internalLogPath ) throws IOException
+        {
+            return new StoreLogService(
+                    userLogProvider,
+                    fileSystem, internalLogPath,
+                    internalLogRotationThreshold, internalLogRotationDelay, maxInternalLogArchives, rotationExecutor, rotationListener );
+        }
+    }
 
-    public StoreLogService( LogProvider userLogProvider, FileSystemAbstraction fileSystem, File storeDir, Config config, JobScheduler jobScheduler ) throws IOException
+    public static Builder withUserLogProvider( LogProvider userLogProvider )
     {
-        this( userLogProvider, fileSystem, storeDir, config, 0L, 0, 0, jobScheduler, Consumers.<LogProvider>noop() );
+        return new Builder().withUserLogProvider( userLogProvider );
     }
 
-    public StoreLogService( LogProvider userLogProvider, FileSystemAbstraction fileSystem, File storeDir, Config config,
-                            long internalLogRotationThreshold, int internalLogRotationDelay, int maxInternalLogArchives,
-                            JobScheduler jobScheduler, final Consumer<LogProvider> rotationListener ) throws IOException
+    public static Builder withRotation( long internalLogRotationThreshold, int internalLogRotationDelay, int maxInternalLogArchives, JobScheduler jobScheduler )
     {
-        this( userLogProvider, fileSystem, storeDir, config, internalLogRotationThreshold, internalLogRotationDelay, maxInternalLogArchives,
-                jobScheduler.executor( JobScheduler.Groups.internalLogRotation ), rotationListener );
+        return new Builder().withRotation( internalLogRotationThreshold, internalLogRotationDelay, maxInternalLogArchives, jobScheduler );
     }
 
-    public StoreLogService( LogProvider userLogProvider, FileSystemAbstraction fileSystem, File storeDir, Config config,
-                            long internalLogRotationThreshold, int internalLogRotationDelay, int maxInternalLogArchives,
-                            Executor rotationExecutor, final Consumer<LogProvider> rotationListener ) throws IOException
+    public static StoreLogService inStoreDirectory( FileSystemAbstraction fileSystem, File storeDir ) throws IOException
```

Comment:
that is a good example: we need StoreLogService to be constructed as fast and as clean as possible so yes we need to keep factory methods that result in logService itself, but not methods that will return builder. 
API users will have ability use or builder or directly exposed factory methods that will return keep things more clean and short.
And that i suppose we can apply to all new builders that coming. What do you think?